src/develop/blend.c

```diff
@@ -2426,6 +2503,331 @@ static void _blend_HSV_color(const _blend_buffer_desc_t *bd, const float *a, flo
   }
 }
 
+/* blend only R-channel in RGB color space without any clamping (a noop for
+ * other color spaces) */
+static void _blend_RGB_R(const _blend_buffer_desc_t *bd, const float *a, float *b, const float *mask,
+                         int flag)
+{
+  float max[4] = { 0 }, min[4] = { 0 };
+  _blend_colorspace_channel_range(bd->cst, min, max);
+
+  if(bd->cst == iop_cs_Lab)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
```

Comment:
Well, we would need several no-op inlines for different color spaces RGB/Lab vs. RAW and for clamping and non-clamping blend modes. A multitude of inlines wouldn't improve things IMHO.